From: Arnd Bergmann <arnd@arndb.de>
Date: Tue, 8 Nov 2016 14:48:59 +0100
Subject: dmaengine: qcom_hidma: hide MSI handler when unused

Git-commit: 8cc12b26b403611175c4f1aec13abc4e383897fb
Patch-mainline: v4.10-rc1
References: fate#320512

The newly added MSI support causes a harmless warning when MSI
is disabled:

drivers/dma/qcom/hidma.c:558:20: error: 'hidma_chirq_handler_msi' defined but not used [-Werror=unused-function]

This adds another #ifdef to match that around the users of the function.

Fixes: 1c0e3e82a7fb ("dmaengine: qcom_hidma: add MSI support for interrupts")
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Acked-by: Sinan Kaya <okaya@codeaurora.org>
Signed-off-by: Vinod Koul <vinod.koul@intel.com>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/dma/qcom/hidma.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/dma/qcom/hidma.c b/drivers/dma/qcom/hidma.c
index d5e7991ad737..10425e134e06 100644
--- a/drivers/dma/qcom/hidma.c
+++ b/drivers/dma/qcom/hidma.c
@@ -555,6 +555,7 @@ static irqreturn_t hidma_chirq_handler(int chirq, void *arg)
 	return hidma_ll_inthandler(chirq, lldev);
 }
 
+#ifdef CONFIG_GENERIC_MSI_IRQ_DOMAIN
 static irqreturn_t hidma_chirq_handler_msi(int chirq, void *arg)
 {
 	struct hidma_lldev **lldevp = arg;
@@ -563,6 +564,7 @@ static irqreturn_t hidma_chirq_handler_msi(int chirq, void *arg)
 	return hidma_ll_inthandler_msi(chirq, *lldevp,
 				       1 << (chirq - dmadev->msi_virqbase));
 }
+#endif
 
 static ssize_t hidma_show_values(struct device *dev,
 				 struct device_attribute *attr, char *buf)
-- 
2.11.0

